guides/advanced_guides/configuration.md

```diff
@@ -287,39 +295,43 @@ This command will stop the process if:
 
 If this command is not called, no snapshot will be imported.
 
-### Ignore missing snapshot
+### Ignore Missing Snapshot
 
 **Environment variable**: `MEILI_IGNORE_MISSING_SNAPSHOT`
 **CLI option**: `--ignore-missing-snapshot`
 
 The engine ignores missing snapshots and does not throw an error in this case.
 
-### Ignore snapshot if db exists
+Requires `--import-snapshot` to be defined.
```

Comment:
Does this mean an error is raised if the flag is given but `--import-snapshot` is not?

Reply:
yep

Reply:
Is this not considered like `--snapshot-path` as a setting that can be pre-added as an ENV variable but will only work once we add `--schedule-snapshot`? What is the idea behind this decision?

Reply:
good point 🤔 

I think there is no reason to persist env_var for theses options:
- `--import-snapshot`
- `--import-dump`
- `--ignore-missing-snapshot`
- `--ignore-snapshot-if-db-exists`

 no?

Reply:
Mmmh, I see this more in a option file context. Where I would only have to add `--import-snapshot` when you need to quickly (or not) load a snapshot, and not all 5 options. But I also understand your point of view. @curquiza @eskombro @MarinPostma any input?

Reply:
> I think there is no reason to persist env_var for theses options:

>    --import-snapshot
>    --import-dump
>    --ignore-missing-snapshot
>    --ignore-snapshot-if-db-exists

> no?

I don't know what to think. I have kind of the same feeling, but I'm not sure at all, I'm not used to creating devops files with a lot of env var...etc
I a

Reply:
I think it's a good choice if two variables/options are meant to be used together, and you use just one, this shouldn't be just ignored without telling the user, and with this in mind, I agree with the requirement of `--import-snapshot` if you use `--ignore-missing-snapshot`. In the other hand, tehre are other cases where options are ignored, so just doing it here doesn't feel totally consistent.


I dont understand what you mean exactly @LegendreM by `I think there is no reason to persist env_var` so sorry if my answer is not pertnent.

Reply:
Could we consider a warning instead of an error?

Reply:
I like the warning idea 👍

Reply:
I have been thinking about this. `--ignore-missing-snapshot` is an opt-in flag, I think that having an environement variable that sets this variable to always true is error prone. We should not design an API that leads our users to commit eventual error (it's very easy to forget that a variable in the environment has be set). Also, this flag makes not sense on it's own, and only makes sense at startup (unlike `--snapshot-dir`). The user shouldn't be allowed to launch a program with an invalid configuration (even if it appears like it does nothing bad). For this reason, I think this should be a hard error, and there should't be an env variable for this one, so previous configuration does not interfere with the current one.

Reply:
Agree with @MarinPostma POV!

So IMO we should keep the hard error as he says, and remove the env var as @LegendreM proposed 👌

Reply:
removed `ENV_VAR` for:

-    --import-snapshot
-    --import-dump
-   --ignore-missing-snapshot
-    --ignore-snapshot-if-db-exists

Reply:
What does this imply? I have a hard time understanding.

Reply:
@LegendreM explained it to me. So now all our options have an env variable alternative except those 4? That is not really consistent...

Reply:
I think that `ENV_VAR` should not describe a starting behavior and should not be conflictual:
- if you both set `MEILI_IMPORT_DUMP` and `MEILI_IMPORT_SNAPSHOT`, it will conflict
- if you  set `MEILI_IMPORT_DUMP` you'll risk to delete data at the next start forgetting that you set this `ENV_VAR`

Reply:
Oke, I understand. Apparently I was the one that did not understand what Marin said and everyone seems to agree. So lets go for that.

But this does not answer my original question: 
`where if you use --ignore-missing-snapshot without --import-snapshot it should not raise an error but a warning.`

Do we agree on this ?

Reply:
I don't like warnings

Reply:
This discussion continues here: 
https://github.com/meilisearch/MeiliSearch/issues/1033

I tried to gather all opinions.